pkg/apis/control/v1/extensions.go

```diff
@@ -0,0 +1,89 @@
+package v1
+
+import (
+	"encoding/hex"
+	"hash"
+
+	"golang.org/x/crypto/blake2b"
+	"golang.org/x/exp/slices"
+)
+
+func (m *PluginManifest) DigestSet() map[string]struct{} {
+	hm := map[string]struct{}{}
+	for _, v := range m.Items {
+		hm[v.GetDigest()] = struct{}{}
+	}
+	return hm
+}
+
+func (m *PluginManifest) PluginDigests() map[string]string {
+	hm := map[string]string{}
+	for _, v := range m.Items {
+		hm[v.GetId()] = v.GetDigest()
+	}
+	return hm
+}
+
+func (m *PluginManifestEntry) GetId() string {
+	return m.GetModule()
+}
+
+func (m *PluginManifestEntry) DigestBytes() []byte {
+	decoded, _ := hex.DecodeString(m.GetDigest())
+	return decoded
+}
+
+func (m *PluginManifestEntry) DigestHash() hash.Hash {
+	h, _ := blake2b.New256(nil)
+	return h
+}
+
+func (m *PluginManifest) Sort() {
+	slices.SortFunc(m.Items, func(a, b *PluginManifestEntry) bool {
+		return a.GetModule() < b.GetModule()
+	})
+}
+
+func (m *PluginArchive) Sort() {
+	slices.SortFunc(m.Items, func(a, b *PluginArchiveEntry) bool {
+		return a.GetMetadata().GetModule() < b.GetMetadata().GetModule()
+	})
+}
+
+func (a *PluginArchive) ToManifest() *PluginManifest {
+	manifest := &PluginManifest{}
+	for _, entry := range a.Items {
+		manifest.Items = append(manifest.Items, entry.Metadata)
+	}
+	return manifest
```

Comment:
Should we be sorting the manifests here?

Reply:
Good question. Could go either way but having it preserve the same order would likely be the least surprising/error-prone.